src/erlcloud_aws.erl

```diff
@@ -410,6 +410,18 @@ auto_config_profile( ProfileOptions ) ->
     Profile = proplists:get_value( profile, ProfileOptions, default ),
     case profile( Profile, ProfileOptions ) of
         {ok, _Config} = Result -> Result;
+        {error, _} -> auto_config_task_metadata()
+    end.
+
+auto_config_task_metadata() ->
+    case erlcloud_ecs_container_credentials:task_credentials_available() of
```

Comment:
not sure why we need this wrapper and not just call `config_task_metadata()` right away.